recipes-security/optee-imx/optee-client-fslc.inc

```diff
@@ -1,42 +1,43 @@
-# Copyright 2017-2024 NXP
+# Copied from meta-arm/recipes-security/optee/optee-client.inc.
+# See: https://github.com/nxp-imx/imx-manifest/blob/imx-linux-scarthgap/imx-6.6.52-2.2.0.xml#L30
+
+SUMMARY = "OP-TEE Client API"
+DESCRIPTION = "Open Portable Trusted Execution Environment - Normal World Client side of the TEE"
+HOMEPAGE = "https://www.op-tee.org/"
 
-SUMMARY = "OPTEE Client libs"
-HOMEPAGE = "http://www.optee.org/"
 LICENSE = "BSD-2-Clause"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=69663ab153298557a59c67a60a743e5b"
 
-SRC_URI = "file://tee-supplicant.service"
-
-S = "${WORKDIR}/git"
-B = "${WORKDIR}/build"
+inherit systemd update-rc.d cmake
 
-inherit python3native systemd features_check pkgconfig
-
-DEPENDS = "util-linux-libuuid"
+SRC_URI = " \
+    git://github.com/OP-TEE/optee_client.git;branch=master;protocol=https \
+    file://tee-supplicant@.service \
+    file://tee-supplicant.sh \
+"
 
-REQUIRED_MACHINE_FEATURES = "optee"
+UPSTREAM_CHECK_GITTAGREGEX = "^(?P<pver>\d+(\.\d+)+)$"
 
-SYSTEMD_SERVICE:${PN} = "tee-supplicant.service"
+S = "${WORKDIR}/git"
 
-EXTRA_OEMAKE = " \
-	-C ${S} O=${B} \
+EXTRA_OECMAKE = " \
+    -DBUILD_SHARED_LIBS=ON \
+    -DCFG_TEE_FS_PARENT_PATH='${localstatedir}/lib/tee' \
 "
+EXTRA_OECMAKE:append:toolchain-clang = " -DCFG_WERROR=0"
 
-do_install () {
-	oe_runmake -C ${S} install
-
-	install -D -p -m0644 ${B}/export/usr/lib/libteec.so.2.0.0 ${D}${libdir}/libteec.so.2.0.0
-	ln -sf libteec.so.2.0.0 ${D}${libdir}/libteec.so.2
-	ln -sf libteec.so.2.0.0 ${D}${libdir}/libteec.so
-
-	install -D -p -m0644 ${B}/export/usr/lib/libckteec.so.0.1.0 ${D}${libdir}/libckteec.so.0.1.0
-	ln -sf libckteec.so.0.1.0 ${D}${libdir}/libckteec.so.0
-	ln -sf libckteec.so.0.1.0 ${D}${libdir}/libckteec.so
+do_install:append() {
+    install -D -p -m0644 ${UNPACKDIR}/tee-supplicant@.service ${D}${systemd_system_unitdir}/tee-supplicant@.service
+    install -D -p -m0755 ${UNPACKDIR}/tee-supplicant.sh ${D}${sysconfdir}/init.d/tee-supplicant
```

Comment:
@eghidoli 

Is UNPACKDIR supported in Scarthgap?

I might be wrong, but I think this will fail even at build time.